src/internal/native/array.ts

```diff
@@ -4,6 +4,23 @@
 import { ImageMagick } from '../../image-magick';
 import { quantumArray } from '@dlemstra/magick-native/magick';
 
+/** @internal */
+export function _withArray<TReturnType>(array: Uint8Array | Uint8ClampedArray, func: (instance: number) => TReturnType): TReturnType {
+    const length = array.length;
+    if (length === 0)
+        return func(0);
+
+    const instance = ImageMagick._api._malloc(length);
```

Comment:
I think we need to do this inside the try because `ImageMagick._api._malloc(length)` might throw an error.